### ISSUE TITLE
Error asking for input: no available tty

### ISSUE THREAD

**Issue by alexandernst** [hashicorp/packer#8682]
#### Overview of the Issue

I'm not able to build an AWS AMI using a very simple packer config/template. The error seems to be caused by the upgrade of the go-tty version in 1.5.1 (¿?).

As extra data, the same config used to work fine in Packer 1.4.5

**Error output**
```
==> amazon-ebs: Provisioning with shell script: /tmp/packer-shell797744527
224 2020/02/03 00:49:28 packer-provisioner-shell plugin: Opening /tmp/packer-shell797744527 for reading
225 2020/02/03 00:49:28 packer-provisioner-shell plugin: [INFO] 392 bytes written for 'uploadData'
226 2020/02/03 00:49:28 [INFO] 392 bytes written for 'uploadData'
227 2020/02/03 00:49:28 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
228 2020/02/03 00:49:28 packer-builder-amazon-ebs plugin: [DEBUG] Starting remote scp process:  scp -vt /tmp
229 2020/02/03 00:49:28 packer-builder-amazon-ebs plugin: [DEBUG] Started SCP session, beginning transfers...
230 2020/02/03 00:49:28 packer-builder-amazon-ebs plugin: [DEBUG] Copying input data into temporary file so we can read the length
231 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading script_3508.sh: perms=C0644 size=392
232 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [DEBUG] SCP session complete, closing stdin pipe.
233 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [DEBUG] Waiting for SSH session to complete.
234 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [DEBUG] scp stderr (length 31): Sink: C0644 392 script_3508.sh
235 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
236 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [DEBUG] starting remote command: chmod 0755 /tmp/script_3508.sh
237 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [INFO] RPC endpoint: Communicator ended with: 0
238 2020/02/03 00:49:29 [INFO] RPC client: Communicator ended with: 0
239 2020/02/03 00:49:29 [INFO] RPC endpoint: Communicator ended with: 0
240 2020/02/03 00:49:29 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 0
241 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
242 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [DEBUG] starting remote command: chmod +x /tmp/script_3508.sh; PACKER_BUILDER_TYPE='amazon-ebs' PACKER_BUILD_NAME='amazon-ebs'  /tmp/script_3508.sh
243 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [ERROR] Remote command exited with '1': chmod +x /tmp/script_3508.sh; PACKER_BUILDER_TYPE='amazon-ebs' PACKER_BUILD_NAME='amazon-ebs'  /tmp/script_3508.sh
244 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [INFO] RPC endpoint: Communicator ended with: 1
245 2020/02/03 00:49:29 [INFO] 0 bytes written for 'stderr'
246 2020/02/03 00:49:29 [INFO] 0 bytes written for 'stdout'
247 2020/02/03 00:49:29 [INFO] RPC client: Communicator ended with: 1
248 2020/02/03 00:49:29 [INFO] RPC endpoint: Communicator ended with: 1
249 2020/02/03 00:49:29 packer-provisioner-shell plugin: [INFO] 0 bytes written for 'stdout'
250 2020/02/03 00:49:29 packer-provisioner-shell plugin: [INFO] 0 bytes written for 'stderr'
251 2020/02/03 00:49:29 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 1
252 2020/02/03 00:49:29 [INFO] (telemetry) ending shell
253 ==> amazon-ebs: Provisioning step had errors: Running the cleanup provisioner, if present...
254 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
255 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
256 ==> amazon-ebs: Terminating the source AWS instance...
257 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
258 2020/02/03 00:49:59 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
259 ==> amazon-ebs: Cleaning up any extra volumes...
260 ==> amazon-ebs: No volumes to clean up, skipping
261 2020/02/03 00:49:59 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
262 2020/02/03 00:49:59 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
263 ==> amazon-ebs: Deleting temporary security group...
264 2020/02/03 00:50:00 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
265 ==> amazon-ebs: Deleting temporary keypair...
266 2020/02/03 00:50:00 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
267 2020/02/03 00:50:00 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
268 2020/02/03 00:50:00 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
269 2020/02/03 00:50:00 [INFO] (telemetry) ending amazon-ebs
270 2020/02/03 00:50:00 Waiting on builds to complete...
271 2020/02/03 00:50:00 machine readable: error-count []string{"1"}
272 ==> Some builds didn't complete successfully and had errors:
273 2020/02/03 00:50:00 machine readable: amazon-ebs,error []string{"Script exited with non-zero exit status: 1.Allowed exit codes are: [0]"}
274 ==> Builds finished but no artifacts were created.
275 2020/02/03 00:50:00 Cancelling builder after context cancellation context canceled
276 2020/02/03 00:50:00 [INFO] (telemetry) Finalizing.
277 Build 'amazon-ebs' errored: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
278 ==> Some builds didn't complete successfully and had errors:
279 --> amazon-ebs: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
280 ==> Builds finished but no artifacts were created.
```

**Packer template**
```
{
    "builders": [
        {
            "type": "amazon-ebs",
            "vault_aws_engine": {
                "name": "*******"
            },
            "region": "us-east-1",
            "source_ami_filter": {
                "filters": {
                    "virtualization-type": "hvm",
                    "name": "********",  // <--- try any debian 10 AMI
                    "root-device-type": "ebs"
                },
                "owners": ["*******"],
                "most_recent": true
            },
            "instance_type": "t3.xlarge",
            "vpc_id": "*******",
            "subnet_id": "*******",
            "ssh_username": "admin",
            "ami_name": "*******",
            "ami_description": "",
            "force_delete_snapshot": true,
            "force_deregister": true,
            "spot_price": "auto",
            "shutdown_behavior": "terminate"
        }
    ],

    "provisioners": [
        {
            "type": "file",
            "source": "scripts",
            "destination": "/tmp/"
        },
        {
            "type": "shell",
            "expect_disconnect": "true",
            "inline": [
                "sudo chmod +x /tmp/scripts/*.sh",
            ]
        }
    ]
}
```

#### Reproduction Steps

Try to build an AWS AMI using provided config.

### Packer version

v.1.5.1

### Operating system and Environment details

The packer itself is ran on an Alpine Arch x64 docker image (but I doubt this can be affecting in any way).

**Comment by sylviamoss** [comment]
Hi there! Thanks for reaching out. We’ll take a look at this when we get a chance.

**Comment by alexandernst** [comment]
I tried downgrading packer to 1.4.5 and I'm getting the same result, which means that there's something strange going on. Is there any way I can further debug this in order to help you?

**Comment by sylviamoss** [comment]
By looking at your logs I would say that running the command `sudo chmod +x /tmp/scripts/*.sh` is not necessary because some lines before that, the permissions for the file are already being set to 0755 which is the same thing as +x.
See: 
`236 2020/02/03 00:49:29 packer-builder-amazon-ebs plugin: [DEBUG] starting remote command: chmod 0755 /tmp/script_3508.sh`
If you remove the shell provisioner, does the template works as expected?

**Comment by alexandernst** [comment]
I tried your suggestions and I got:

```
181 ==> amazon-ebs: Uploading scripts => /tmp/
182 2020/02/03 11:12:15 packer-builder-amazon-ebs plugin: [DEBUG] Upload dir 'scripts' to '/tmp/'
183 2020/02/03 11:12:15 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
184 2020/02/03 11:12:15 packer-builder-amazon-ebs plugin: [DEBUG] Starting remote scp process:  scp -rvt /tmp/
185 2020/02/03 11:12:15 packer-builder-amazon-ebs plugin: [DEBUG] Started SCP session, beginning transfers...
186 2020/02/03 11:12:15 packer-builder-amazon-ebs plugin: [DEBUG] No trailing slash, creating the source directory name
187 2020/02/03 11:12:15 packer-builder-amazon-ebs plugin: [DEBUG] SCP: starting directory upload: scripts
188 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading install_cli_tools.sh: perms=C0666 size=667
189 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading harden_kernel.sh: perms=C0666 size=4118
190 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading install_mta.sh: perms=C0666 size=130
191 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_netdata: perms=C0666 size=351
192 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_cron: perms=C0666 size=1054
193 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading install_autorun.sh: perms=C0666 size=808
194 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading end.sh: perms=C0666 size=382
195 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading stop_apt_daily.sh: perms=C0666 size=388
196 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading sec.sh: perms=C0666 size=15806
197 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_timezone: perms=C0666 size=174
198 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading init.sh: perms=C0666 size=651
199 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_hostname: perms=C0666 size=621
200 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading opts.sh: perms=C0666 size=4728
201 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_notifications: perms=C0666 size=537
202 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] SCP session complete, closing stdin pipe.
203 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] Waiting for SSH session to complete.
204 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp stderr (length 474): Sink: D0777 0 scripts
205 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 667 install_cli_tools.sh
206 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 4118 harden_kernel.sh
207 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 130 install_mta.sh
208 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 351 configure_netdata
209 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 1054 configure_cron
210 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 808 install_autorun.sh
211 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 382 end.sh
212 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 388 stop_apt_daily.sh
213 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 15806 sec.sh
214 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 174 configure_timezone
215 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 651 init.sh
216 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 621 configure_hostname
217 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 4728 opts.sh
218 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: C0666 537 configure_notifications
219 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: Sink: E
220 2020/02/03 11:12:16 [INFO] (telemetry) ending file
221 2020/02/03 11:12:16 [INFO] (telemetry) Starting provisioner shell
222 2020/02/03 11:12:16 Error asking for input: no available tty
223 ==> amazon-ebs: Provisioning with shell script: /tmp/packer-shell986672075
224 2020/02/03 11:12:16 packer-provisioner-shell plugin: Opening /tmp/packer-shell986672075 for reading
225 2020/02/03 11:12:16 packer-provisioner-shell plugin: [INFO] 360 bytes written for 'uploadData'
226 2020/02/03 11:12:16 [INFO] 360 bytes written for 'uploadData'
227 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
228 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] Starting remote scp process:  scp -vt /tmp
229 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] Started SCP session, beginning transfers...
230 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] Copying input data into temporary file so we can read the length
231 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading script_6802.sh: perms=C0644 size=360
232 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] SCP session complete, closing stdin pipe.
233 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] Waiting for SSH session to complete.
234 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] scp stderr (length 31): Sink: C0644 360 script_6802.sh
235 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
236 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] starting remote command: chmod 0755 /tmp/script_6802.sh
237 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [INFO] RPC endpoint: Communicator ended with: 0
238 2020/02/03 11:12:16 [INFO] RPC client: Communicator ended with: 0
239 2020/02/03 11:12:16 [INFO] RPC endpoint: Communicator ended with: 0
240 2020/02/03 11:12:16 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 0
241 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
242 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [DEBUG] starting remote command: chmod +x /tmp/script_6802.sh; PACKER_BUILDER_TYPE='amazon-ebs' PACKER_BUILD_NAME='amazon-ebs'  /tmp/script_6802.sh
243 ==> amazon-ebs: sudo: /tmp/scripts/stop_apt_daily.sh: command not found
244 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [ERROR] Remote command exited with '1': chmod +x /tmp/script_6802.sh; PACKER_BUILDER_TYPE='amazon-ebs' PACKER_BUILD_NAME='amazon-ebs'  /tmp/script_6802.sh
245 2020/02/03 11:12:16 packer-builder-amazon-ebs plugin: [INFO] RPC endpoint: Communicator ended with: 1
246 2020/02/03 11:12:16 [INFO] 0 bytes written for 'stdout'
247 2020/02/03 11:12:16 [INFO] 56 bytes written for 'stderr'
248 2020/02/03 11:12:16 [INFO] RPC client: Communicator ended with: 1
249 2020/02/03 11:12:16 [INFO] RPC endpoint: Communicator ended with: 1
250 2020/02/03 11:12:16 packer-provisioner-shell plugin: [INFO] 0 bytes written for 'stdout'
251 2020/02/03 11:12:16 packer-provisioner-shell plugin: [INFO] 56 bytes written for 'stderr'
252 2020/02/03 11:12:16 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 1
253 2020/02/03 11:12:16 [INFO] (telemetry) ending shell
254 ==> amazon-ebs: Provisioning step had errors: Running the cleanup provisioner, if present...
```

I really don't think that my `sudo chmod +x /tmp/scripts/*.sh` has anything to do with the `chmod 0755 /tmp/script_3508.sh`. I think that the `chmod 0755 /tmp/script_3508.sh` is being ran by packer in order to be able to run my commands, which I guess are stored in a file called `script_3508.sh`.

Also, `sudo chmod +x /tmp/scripts/*.sh` really shouldn't fail in any way (necessary or not).

**Comment by sylviamoss** [comment]
Yes, you're right! It shouldn't fail at all. I'm trying to think of a workaround solution so this wouldn't keep you from creating the image. 
I need to go into this with more details but right now I'm working with something else. I will come back later to try to figure this out. OK? 
Meanwhile, let me know if you figure it out by yourself!

**Comment by alexandernst** [comment]
Okey! I'll try to test some stuff and see if I can see what exactly is causing the crash, but I must say that I'm absolutely lost as this code has been working for months, and now it doesn't work with neither 1.4.5 nor 1.5.1.

Ping me if you need me to run more stuff for you!

Thank you!

**Comment by alexandernst** [comment]
I have been debugging this further. I added

```
            ...
            "inline": [
                "ls -la /tmp/",
                "ls -la /tmp/scripts/",
                "xx=$(sudo chmod +x /tmp/scripts/*.sh 2>&1; true)",
                "echo $xx",

                "sudo chmod +x /tmp/scripts/*.sh",
                ...
```

and I'm getting this:

```
242 2020/02/03 12:51:26 packer-builder-amazon-ebs plugin: [DEBUG] starting remote command: chmod +x /tmp/script_9998.sh; PACKER_BUILDER_TYPE='amazon-ebs' PACKER_BUILD_NAME='amazon-ebs'  /tmp/script_9998.sh
243     amazon-ebs: total 40
244     amazon-ebs: drwxrwxrwt  9 root  root  4096 Feb  3 12:51 .
245     amazon-ebs: drwxr-xr-x 19 root  root  4096 Feb  3 12:51 ..
246     amazon-ebs: drwxrwxrwt  2 root  root  4096 Feb  3 12:51 .ICE-unix
247     amazon-ebs: drwxrwxrwt  2 root  root  4096 Feb  3 12:51 .Test-unix
248     amazon-ebs: drwxrwxrwt  2 root  root  4096 Feb  3 12:51 .X11-unix
249     amazon-ebs: drwxrwxrwt  2 root  root  4096 Feb  3 12:51 .XIM-unix
250     amazon-ebs: drwxrwxrwt  2 root  root  4096 Feb  3 12:51 .font-unix
251     amazon-ebs: -rwxr-xr-x  1 admin admin  484 Feb  3 12:51 script_9998.sh
252     amazon-ebs: drwxr-xr-x  2 admin admin 4096 Feb  3 12:51 scripts
253     amazon-ebs: drwx------  3 root  root  4096 Feb  3 12:51 systemd-private-c4065131390146c49593b9bf1ec73033-chrony.service-dEylxi
254     amazon-ebs: total 84
255     amazon-ebs: drwxr-xr-x 2 admin admin  4096 Feb  3 12:51 .
256     amazon-ebs: drwxrwxrwt 9 root  root   4096 Feb  3 12:51 ..
257     amazon-ebs: -rw-r--r-- 1 admin admin  1054 Feb  3 12:51 configure_cron
258     amazon-ebs: -rw-r--r-- 1 admin admin   621 Feb  3 12:51 configure_hostname
259     amazon-ebs: -rw-r--r-- 1 admin admin   351 Feb  3 12:51 configure_netdata
260     amazon-ebs: -rw-r--r-- 1 admin admin   537 Feb  3 12:51 configure_notifications
261     amazon-ebs: -rw-r--r-- 1 admin admin   174 Feb  3 12:51 configure_timezone
262     amazon-ebs: -rw-r--r-- 1 admin admin   382 Feb  3 12:51 end.sh
263     amazon-ebs: -rw-r--r-- 1 admin admin  4118 Feb  3 12:51 harden_kernel.sh
264     amazon-ebs: -rw-r--r-- 1 admin admin   651 Feb  3 12:51 init.sh
265     amazon-ebs: -rw-r--r-- 1 admin admin   808 Feb  3 12:51 install_autorun.sh
266     amazon-ebs: -rw-r--r-- 1 admin admin   667 Feb  3 12:51 install_cli_tools.sh
267     amazon-ebs: -rw-r--r-- 1 admin admin   130 Feb  3 12:51 install_mta.sh
268     amazon-ebs: -rw-r--r-- 1 admin admin  4728 Feb  3 12:51 opts.sh
269     amazon-ebs: -rw-r--r-- 1 admin admin 15806 Feb  3 12:51 sec.sh
270     amazon-ebs: -rw-r--r-- 1 admin admin   388 Feb  3 12:51 stop_apt_daily.sh
271     amazon-ebs:
272 2020/02/03 12:51:26 packer-builder-amazon-ebs plugin: [ERROR] Remote command exited with '1': chmod +x /tmp/script_9998.sh; PACKER_BUILDER_TYPE='amazon-ebs' PACKER_BUILD_NAME='amazon-ebs'  /tmp/script_9998.sh
```

Which means that packer is actually running my commands. First I see the `ls -la /tmp/`, then I see the `ls -la /tmp/scripts/`. Then packer tries to run the `xx=$(sudo chmod +x /tmp/scripts/*.sh 2>&1; true)`, which will catch all the output (including errors) and save it to a variable called `xx`. Then I see that packer is actually printing the `xx` value, which is an empty line (line 271), but for some reason the next command (`chmod +x ...`) fails with an exit code of 1.

**Comment by alexandernst** [comment]
More debugging:

```
164 ==> amazon-ebs: Using ssh communicator to connect: 172.16.19.109
165 2020/02/03 14:22:04 packer-builder-amazon-ebs plugin: [INFO] Waiting for SSH, up to timeout: 5m0s
166 ==> amazon-ebs: Waiting for SSH to become available...
167 2020/02/03 14:22:04 packer-builder-amazon-ebs plugin: Using ssh_host value: 172.16.19.109
168 2020/02/03 14:22:19 packer-builder-amazon-ebs plugin: [DEBUG] TCP connection to SSH ip/port failed: dial tcp 172.16.19.109:22: i/o timeout
169 2020/02/03 14:22:24 packer-builder-amazon-ebs plugin: Using ssh_host value: 172.16.19.109
170 2020/02/03 14:22:27 packer-builder-amazon-ebs plugin: [DEBUG] TCP connection to SSH ip/port failed: dial tcp 172.16.19.109:22: connect: connection refused
171 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Using ssh_host value: 172.16.19.109
172 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [INFO] Attempting SSH connection to 172.16.19.109:22...
173 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] reconnecting to TCP connection for SSH
174 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] handshaking with SSH
175 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] handshake complete!
176 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [INFO] no local agent socket, will not connect agent
177 ==> amazon-ebs: Connected to SSH!
178 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
179 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Running the provision hook
180 2020/02/03 14:22:32 [INFO] (telemetry) Starting provisioner file
181 2020/02/03 14:22:32 Error asking for input: no available tty
182 ==> amazon-ebs: Uploading scripts => /tmp/
183 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Upload dir 'scripts' to '/tmp/'
184 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
185 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Starting remote scp process:  scp -rvt /tmp/
186 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Started SCP session, beginning transfers...
187 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] No trailing slash, creating the source directory name
188 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] SCP: starting directory upload: scripts
189 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading install_cli_tools.sh: perms=C0666 size=667
190 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading harden_kernel.sh: perms=C0666 size=4118
191 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading install_mta.sh: perms=C0666 size=130
192 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_netdata: perms=C0666 size=351
193 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_cron: perms=C0666 size=1054
194 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading install_autorun.sh: perms=C0666 size=808
195 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading end.sh: perms=C0666 size=382
196 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading stop_apt_daily.sh: perms=C0666 size=388
197 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading sec.sh: perms=C0666 size=15806
198 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_timezone: perms=C0666 size=174
199 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading init.sh: perms=C0666 size=651
200 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_hostname: perms=C0666 size=621
201 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading opts.sh: perms=C0666 size=4728
202 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading configure_notifications: perms=C0666 size=537
203 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] SCP session complete, closing stdin pipe.
204 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Waiting for SSH session to complete.
205 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp stderr (length 474): Sink: D0777 0 scripts
206 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 667 install_cli_tools.sh
207 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 4118 harden_kernel.sh
208 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 130 install_mta.sh
209 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 351 configure_netdata
210 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 1054 configure_cron
211 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 808 install_autorun.sh
212 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 382 end.sh
213 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 388 stop_apt_daily.sh
214 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 15806 sec.sh
215 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 174 configure_timezone
216 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 651 init.sh
217 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 621 configure_hostname
218 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 4728 opts.sh
219 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: C0666 537 configure_notifications
220 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: Sink: E
221 2020/02/03 14:22:32 [INFO] (telemetry) ending file
222 2020/02/03 14:22:32 [INFO] (telemetry) Starting provisioner shell
223 2020/02/03 14:22:32 Error asking for input: no available tty
224 ==> amazon-ebs: Provisioning with shell script: /tmp/packer-shell077199658
225 2020/02/03 14:22:32 packer-provisioner-shell plugin: Opening /tmp/packer-shell077199658 for reading
226 2020/02/03 14:22:32 packer-provisioner-shell plugin: [INFO] 521 bytes written for 'uploadData'
227 2020/02/03 14:22:32 [INFO] 521 bytes written for 'uploadData'
228 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
229 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Starting remote scp process:  scp -vt /tmp
230 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Started SCP session, beginning transfers...
231 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Copying input data into temporary file so we can read the length
232 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp: Uploading script_4193.sh: perms=C0644 size=521
233 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] SCP session complete, closing stdin pipe.
234 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Waiting for SSH session to complete.
235 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] scp stderr (length 31): Sink: C0644 521 script_4193.sh
236 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
237 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] starting remote command: chmod 0755 /tmp/script_4193.sh
238 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [INFO] RPC endpoint: Communicator ended with: 0
239 2020/02/03 14:22:32 [INFO] RPC client: Communicator ended with: 0
240 2020/02/03 14:22:32 [INFO] RPC endpoint: Communicator ended with: 0
241 2020/02/03 14:22:32 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 0
242 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] Opening new ssh session
243 2020/02/03 14:22:32 packer-builder-amazon-ebs plugin: [DEBUG] starting remote command: chmod +x /tmp/script_4193.sh; PACKER_BUILDER_TYPE='amazon-ebs' PACKER_BUILD_NAME='amazon-ebs'  /tmp/script_4193.sh
244     amazon-ebs: Get:1 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
245     amazon-ebs: Get:2 http://cdn-aws.deb.debian.org/debian buster InRelease [122 kB]
246     amazon-ebs: Get:3 http://cdn-aws.deb.debian.org/debian buster-updates InRelease [49.3 kB]
247     amazon-ebs: Get:4 http://cdn-aws.deb.debian.org/debian buster-backports InRelease [46.7 kB]
248     amazon-ebs: Get:5 http://security.debian.org/debian-security buster/updates/main Sources [101 kB]
249     amazon-ebs: Get:6 http://security.debian.org/debian-security buster/updates/main amd64 Packages [175 kB]
250     amazon-ebs: Get:7 http://security.debian.org/debian-security buster/updates/main Translation-en [92.4 kB]
251     amazon-ebs: Get:8 http://cdn-aws.deb.debian.org/debian buster/main Sources [7832 kB]
252     amazon-ebs: Get:9 http://cdn-aws.deb.debian.org/debian buster/main amd64 Packages [7908 kB]
253     amazon-ebs: Get:10 http://cdn-aws.deb.debian.org/debian buster/main Translation-en [5969 kB]
254     amazon-ebs: Get:11 http://cdn-aws.deb.debian.org/debian buster-updates/main Sources [2544 B]
255     amazon-ebs: Get:12 http://cdn-aws.deb.debian.org/debian buster-updates/main amd64 Packages [5792 B]
256     amazon-ebs: Get:13 http://cdn-aws.deb.debian.org/debian buster-updates/main Translation-en [3655 B]
257     amazon-ebs: Get:14 http://cdn-aws.deb.debian.org/debian buster-backports/main amd64 Packages [193 kB]
258     amazon-ebs: Get:15 http://cdn-aws.deb.debian.org/debian buster-backports/main Translation-en [160 kB]
259     amazon-ebs: Fetched 22.7 MB in 4s (6125 kB/s)
260     amazon-ebs: Reading package lists...
261     amazon-ebs: Reading package lists...
262     amazon-ebs: Building dependency tree...
263     amazon-ebs: Reading state information...
264     amazon-ebs: The following additional packages will be installed:
265     amazon-ebs:   libunwind8
266     amazon-ebs: The following NEW packages will be installed:
267     amazon-ebs:   libunwind8 strace
268     amazon-ebs: 0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
269     amazon-ebs: Need to get 948 kB of archives.
270     amazon-ebs: After this operation, 2293 kB of additional disk space will be used.
271     amazon-ebs: Get:1 http://cdn-aws.deb.debian.org/debian buster/main amd64 libunwind8 amd64 1.2.1-9 [50.9 kB]
272     amazon-ebs: Get:2 http://cdn-aws.deb.debian.org/debian buster/main amd64 strace amd64 4.26-0.2 [898 kB]
273 ==> amazon-ebs: debconf: unable to initialize frontend: Dialog
274 ==> amazon-ebs: debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
275 ==> amazon-ebs: debconf: falling back to frontend: Readline
276 ==> amazon-ebs: debconf: unable to initialize frontend: Readline
277 ==> amazon-ebs: debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.28.1 /usr/local/share/perl/5.28.1 /usr/lib/x86_64-linux-gnu/perl5/5.28 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.28 /usr/share/perl/5.28 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7, <> line 2.)
278 ==> amazon-ebs: debconf: falling back to frontend: Teletype
279 ==> amazon-ebs: dpkg-preconfigure: unable to re-open stdin:
280     amazon-ebs: Fetched 948 kB in 0s (31.0 MB/s)
281     amazon-ebs: Selecting previously unselected package libunwind8:amd64.
282     amazon-ebs: (Reading database ... 23445 files and directories currently installed.)
283     amazon-ebs: Preparing to unpack .../libunwind8_1.2.1-9_amd64.deb ...
284     amazon-ebs: Unpacking libunwind8:amd64 (1.2.1-9) ...
285     amazon-ebs: Selecting previously unselected package strace.
286     amazon-ebs: Preparing to unpack .../strace_4.26-0.2_amd64.deb ...
287     amazon-ebs: Unpacking strace (4.26-0.2) ...
288     amazon-ebs: Setting up libunwind8:amd64 (1.2.1-9) ...
289     amazon-ebs: Setting up strace (4.26-0.2) ...
290     amazon-ebs: Processing triggers for libc-bin (2.28-10) ...
291     amazon-ebs: total 84
292     amazon-ebs: drwxr-xr-x 2 admin admin  4096 Feb  3 14:22 .
293     amazon-ebs: drwxrwxrwt 9 root  root   4096 Feb  3 14:22 ..
294     amazon-ebs: -rw-r--r-- 1 admin admin  1054 Feb  3 14:22 configure_cron
295     amazon-ebs: -rw-r--r-- 1 admin admin   621 Feb  3 14:22 configure_hostname
296     amazon-ebs: -rw-r--r-- 1 admin admin   351 Feb  3 14:22 configure_netdata
297     amazon-ebs: -rw-r--r-- 1 admin admin   537 Feb  3 14:22 configure_notifications
298     amazon-ebs: -rw-r--r-- 1 admin admin   174 Feb  3 14:22 configure_timezone
299     amazon-ebs: -rw-r--r-- 1 admin admin   382 Feb  3 14:22 end.sh
300     amazon-ebs: -rw-r--r-- 1 admin admin  4118 Feb  3 14:22 harden_kernel.sh
301     amazon-ebs: -rw-r--r-- 1 admin admin   651 Feb  3 14:22 init.sh
302     amazon-ebs: -rw-r--r-- 1 admin admin   808 Feb  3 14:22 install_autorun.sh
303     amazon-ebs: -rw-r--r-- 1 admin admin   667 Feb  3 14:22 install_cli_tools.sh
304 ==> amazon-ebs: execve("/usr/bin/chmod", ["chmod", "+x", "/tmp/scripts/end.sh", "/tmp/scripts/harden_kernel.sh", "/tmp/scripts/init.sh", "/tmp/scripts/install_autorun.sh", "/tmp/scripts/install_cli_tools.s"..., "/tmp/scripts/install_mta.sh", "/tmp/scripts/opts.sh", "/tmp/scripts/sec.sh", "/tmp/scripts/stop_apt_daily.sh"], 0x7ffe00d032b0 /* 12 vars */) = 0
305 ==> amazon-ebs: brk(NULL)                               = 0x563231945000
306 ==> amazon-ebs: access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
307     amazon-ebs: -rw-r--r-- 1 admin admin   130 Feb  3 14:22 install_mta.sh
308 ==> amazon-ebs: openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
309 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=12792, ...}) = 0
310     amazon-ebs: -rw-r--r-- 1 admin admin  4728 Feb  3 14:22 opts.sh
311 ==> amazon-ebs: mmap(NULL, 12792, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2f16000
312     amazon-ebs: -rw-r--r-- 1 admin admin 15806 Feb  3 14:22 sec.sh
313     amazon-ebs: -rw-r--r-- 1 admin admin   388 Feb  3 14:22 stop_apt_daily.sh
314 ==> amazon-ebs: close(3)                                = 0
315 ==> amazon-ebs: openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
316 ==> amazon-ebs: read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\260A\2\0\0\0\0\0"..., 832) = 832
317 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0755, st_size=1824496, ...}) = 0
318 ==> amazon-ebs: mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f4cc2f14000
319 ==> amazon-ebs: mmap(NULL, 1837056, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f4cc2d53000
320 ==> amazon-ebs: mprotect(0x7f4cc2d75000, 1658880, PROT_NONE) = 0
321 ==> amazon-ebs: mmap(0x7f4cc2d75000, 1343488, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x22000) = 0x7f4cc2d75000
322 ==> amazon-ebs: mmap(0x7f4cc2ebd000, 311296, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x16a000) = 0x7f4cc2ebd000
323 ==> amazon-ebs: mmap(0x7f4cc2f0a000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1b6000) = 0x7f4cc2f0a000
324 ==> amazon-ebs: mmap(0x7f4cc2f10000, 14336, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f4cc2f10000
325 ==> amazon-ebs: close(3)                                = 0
326 ==> amazon-ebs: arch_prctl(ARCH_SET_FS, 0x7f4cc2f15540) = 0
327 ==> amazon-ebs: mprotect(0x7f4cc2f0a000, 16384, PROT_READ) = 0
328 ==> amazon-ebs: mprotect(0x563231491000, 4096, PROT_READ) = 0
329 ==> amazon-ebs: mprotect(0x7f4cc2f41000, 4096, PROT_READ) = 0
330 ==> amazon-ebs: munmap(0x7f4cc2f16000, 12792)           = 0
331 ==> amazon-ebs: brk(NULL)                               = 0x563231945000
332 ==> amazon-ebs: brk(0x563231966000)                     = 0x563231966000
333 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/locale-archive", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
334 ==> amazon-ebs: openat(AT_FDCWD, "/usr/share/locale/locale.alias", O_RDONLY|O_CLOEXEC) = 3
335 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=2995, ...}) = 0
336 ==> amazon-ebs: read(3, "# Locale name alias data base.\n#"..., 4096) = 2995
337 ==> amazon-ebs: read(3, "", 4096)                       = 0
338 ==> amazon-ebs: close(3)                                = 0
339 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_IDENTIFICATION", O_RDONLY|O_CLOEXEC) = 3
340 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=252, ...}) = 0
341 ==> amazon-ebs: mmap(NULL, 252, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2f19000
342 ==> amazon-ebs: close(3)                                = 0
343 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/x86_64-linux-gnu/gconv/gconv-modules.cache", O_RDONLY) = 3
344 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=26402, ...}) = 0
345 ==> amazon-ebs: mmap(NULL, 26402, PROT_READ, MAP_SHARED, 3, 0) = 0x7f4cc2d4c000
346 ==> amazon-ebs: close(3)                                = 0
347 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_MEASUREMENT", O_RDONLY|O_CLOEXEC) = 3
348 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=23, ...}) = 0
349 ==> amazon-ebs: mmap(NULL, 23, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2f18000
350 ==> amazon-ebs: close(3)                                = 0
351 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_TELEPHONE", O_RDONLY|O_CLOEXEC) = 3
352 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=47, ...}) = 0
353 ==> amazon-ebs: mmap(NULL, 47, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2f17000
354 ==> amazon-ebs: close(3)                                = 0
355 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_ADDRESS", O_RDONLY|O_CLOEXEC) = 3
356 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=131, ...}) = 0
357 ==> amazon-ebs: mmap(NULL, 131, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2f16000
358 ==> amazon-ebs: close(3)                                = 0
359 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_NAME", O_RDONLY|O_CLOEXEC) = 3
360 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=62, ...}) = 0
361 ==> amazon-ebs: mmap(NULL, 62, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2d4b000
362 ==> amazon-ebs: close(3)                                = 0
363 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_PAPER", O_RDONLY|O_CLOEXEC) = 3
364 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=34, ...}) = 0
365 ==> amazon-ebs: mmap(NULL, 34, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2d4a000
366 ==> amazon-ebs: close(3)                                = 0
367 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_MESSAGES", O_RDONLY|O_CLOEXEC) = 3
368 ==> amazon-ebs: fstat(3, {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
369 ==> amazon-ebs: close(3)                                = 0
370 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_MESSAGES/SYS_LC_MESSAGES", O_RDONLY|O_CLOEXEC) = 3
371 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=48, ...}) = 0
372 ==> amazon-ebs: mmap(NULL, 48, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2d49000
373 ==> amazon-ebs: close(3)                                = 0
374 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_MONETARY", O_RDONLY|O_CLOEXEC) = 3
375 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=270, ...}) = 0
376 ==> amazon-ebs: mmap(NULL, 270, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2d48000
377 ==> amazon-ebs: close(3)                                = 0
378 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_COLLATE", O_RDONLY|O_CLOEXEC) = 3
379 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=1517626, ...}) = 0
380 ==> amazon-ebs: mmap(NULL, 1517626, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2bd5000
381 ==> amazon-ebs: close(3)                                = 0
382 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_TIME", O_RDONLY|O_CLOEXEC) = 3
383 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=3360, ...}) = 0
384 ==> amazon-ebs: mmap(NULL, 3360, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2bd4000
385 ==> amazon-ebs: close(3)                                = 0
386 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_NUMERIC", O_RDONLY|O_CLOEXEC) = 3
387 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=50, ...}) = 0
388 ==> amazon-ebs: mmap(NULL, 50, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2bd3000
389 ==> amazon-ebs: close(3)                                = 0
390 ==> amazon-ebs: openat(AT_FDCWD, "/usr/lib/locale/C.UTF-8/LC_CTYPE", O_RDONLY|O_CLOEXEC) = 3
391 ==> amazon-ebs: fstat(3, {st_mode=S_IFREG|0644, st_size=200752, ...}) = 0
392 2020/02/03 14:22:40 [INFO] 3883 bytes written for 'stdout'
393 ==> amazon-ebs: mmap(NULL, 200752, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f4cc2ba1000
394 2020/02/03 14:22:40 [INFO] 7967 bytes written for 'stderr'
395 2020/02/03 14:22:40 packer-builder-amazon-ebs plugin: [ERROR] Remote command exited with '1': chmod +x /tmp/script_4193.sh; PACKER_BUILDER_TYPE='amazon-ebs' PACKER_BUILD_NAME='amazon-ebs'  /tmp/script_4193.sh
396 2020/02/03 14:22:40 packer-builder-amazon-ebs plugin: [INFO] RPC endpoint: Communicator ended with: 1
397 2020/02/03 14:22:40 [INFO] RPC client: Communicator ended with: 1
398 2020/02/03 14:22:40 [INFO] RPC endpoint: Communicator ended with: 1
399 2020/02/03 14:22:40 packer-provisioner-shell plugin: [INFO] 3883 bytes written for 'stdout'
400 ==> amazon-ebs: close(3)                                = 0
401 ==> amazon-ebs: umask(000)                              = 022
402 ==> amazon-ebs: stat("/tmp/scripts/end.sh", {st_mode=S_IFREG|0644, st_size=382, ...}) = 0
403 2020/02/03 14:22:40 packer-provisioner-shell plugin: [INFO] 7967 bytes written for 'stderr'
404 2020/02/03 14:22:40 packer-provisioner-shell plugin: [INFO] RPC client: Communicator ended with: 1
405 ==> amazon-ebs: fchmodat(AT_FDCWD, "/tmp/scripts/end.sh", 0755) = 0
406 ==> amazon-ebs: stat("/tmp/scripts/harden_kernel.sh", {st_mode=S_IFREG|0644, st_size=4118, ...}) = 0
407 ==> amazon-ebs: fchmodat(AT_FDCWD, "/tmp/scripts/harden_kernel.sh", 0755) = 0
408 ==> amazon-ebs: stat("/tmp/scripts/init.sh", {st_mode=S_IFREG|0644, st_size=651, ...}) = 0
409 ==> amazon-ebs: fchmodat(AT_FDCWD, "/tmp/scripts/init.sh", 0755) = 0
410 ==> amazon-ebs: stat("/tmp/scripts/install_autorun.sh", {st_mode=S_IFREG|0644, st_size=808, ...}) = 0
411 ==> amazon-ebs: fchmodat(AT_FDCWD, "/tmp/scripts/install_autorun.sh", 0755) = 0
412 ==> amazon-ebs: stat("/tmp/scripts/install_cli_tools.sh", {st_mode=S_IFREG|0644, st_size=667, ...}) = 0
413 ==> amazon-ebs: fchmodat(AT_FDCWD, "/tmp/scripts/install_cli_tools.sh", 0755) = 0
414 ==> amazon-ebs: stat("/tmp/scripts/install_mta.sh", {st_mode=S_IFREG|0644, st_size=130, ...}) = 0
415 ==> amazon-ebs: fchmodat(AT_FDCWD, "/tmp/scripts/install_mta.sh", 0755) = 0
416 ==> amazon-ebs: stat("/tmp/scripts/opts.sh", {st_mode=S_IFREG|0644, st_size=4728, ...}) = 0
417 ==> amazon-ebs: fchmodat(AT_FDCWD, "/tmp/scripts/opts.sh", 0755) = 0
418 ==> amazon-ebs: stat("/tmp/scripts/sec.sh", {st_mode=S_IFREG|0644, st_size=15806, ...}) = 0
419 ==> amazon-ebs: fchmodat(AT_FDCWD, "/tmp/scripts/sec.sh", 0755) = 0
420 ==> amazon-ebs: stat("/tmp/scripts/stop_apt_daily.sh", {st_mode=S_IFREG|0644, st_size=388, ...}) = 0
421 ==> amazon-ebs: fchmodat(AT_FDCWD, "/tmp/scripts/stop_apt_daily.sh", 0755) = 0
422 ==> amazon-ebs: close(1)                                = 0
423 ==> amazon-ebs: close(2)                                = 0
424 ==> amazon-ebs: exit_group(0)                           = ?
425 ==> amazon-ebs: +++ exited with 0 +++
426 2020/02/03 14:22:40 [INFO] (telemetry) ending shell
427 ==> amazon-ebs: Provisioning step had errors: Running the cleanup provisioner, if present...
428 2020/02/03 14:22:40 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
429 2020/02/03 14:22:40 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
430 2020/02/03 14:22:40 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
431 ==> amazon-ebs: Terminating the source AWS instance...
432 ==> amazon-ebs: Cleaning up any extra volumes...
433 2020/02/03 14:23:11 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
434 ==> amazon-ebs: No volumes to clean up, skipping
435 2020/02/03 14:23:11 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
436 2020/02/03 14:23:11 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
437 ==> amazon-ebs: Deleting temporary security group...
438 2020/02/03 14:23:11 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
439 ==> amazon-ebs: Deleting temporary keypair...
440 2020/02/03 14:23:12 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
441 2020/02/03 14:23:12 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
442 2020/02/03 14:23:12 packer-builder-amazon-ebs plugin: Error asking for input: no available tty
443 Build 'amazon-ebs' errored: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
444 2020/02/03 14:23:12 [INFO] (telemetry) ending amazon-ebs
445 2020/02/03 14:23:12 Waiting on builds to complete...
```

This is the strace output of the `chmod` that is failing. The very first thing I'm noticing here is that there is some sort of multi-thread or multi-processing going on, because the output is kind of messed up.

Also, I see that the exit status of strace is 0, so why is packer saying that something exited with an exit code of 1? Maybe there is something running in parallel? 

But why is packer running anything in parallel when I'm running it with `-debug`? Isn't that supposed to limit packer's concurrency to 1?

**Comment by alexandernst** [comment]
I just confirmed that the bug is caused by the last command of the `stop_apt_daily.sh` script, which indeed means that packer is running several scripts at the same time (or at least that is what I understand from the output). 

My problem is solved, but I'm not really sure if you want me to keep this issue open in order to check the multi-processing bug (¿?) or if maybe I should open another issue?

**Comment by sylviamoss** [comment]
Could you test placing your scripts outside `/tmp/` and then run the `chmod` on each one of them explicitly, without using * to run the command on all of them? 
You can do one test for each case. One for placing scripts in another folder and other for changing permissions for each file. 
I think the `/tmp/` folder is used by packer for internal things and maybe this could be confusing.

If the results are the same I think we can close this one and open another one more specific.

Thanks for debugging and going into this!

**Comment by alexandernst** [comment]
Of course!

I ran the chmod on each file and indeed, that was not the problem. The problem was in the `stop_apt_daily.sh` script.

Also, I wasn't copying the scripts in the `/tmp/` folder, but in the `/tmp/scripts/` folder.

No problem! Thanks for the help!
